### PR TITLE
Add support for disabling spell check

### PR DESCRIPTION
**Pull request by SimonBrandner**
Requires https://github.com/matrix-org/matrix-react-sdk/pull/8604
Requires https://github.com/vector-im/element-desktop/pull/379
Notes: none

---
This change has no change notes, so will not be included in the changelog.

### CODE REVIEW

**Review by turt2live — CHANGES_REQUESTED** [review]
this needs a bit of context - there's claims on the linked issues that we can't reasonably manage spell check for mac OS, hence the check. 

Evidence to the contrary would be nice :)

**Comment by SimonBrandner** [comment]
> this needs a bit of context - there's claims on the linked issues that we can't reasonably manage spell check for mac OS, hence the check.
> 
> Evidence to the contrary would be nice :)

The method had a rather unideal name :/

**Review by turt2live — APPROVED** [review]
bonus points if all the functions here get their public/private/protected declarations

**Comment by SimonBrandner** [comment]
https://github.com/vector-im/element-web/issues/22235 :P

**Review by turt2live — APPROVED** [review]

File: src/vector/platform/ElectronPlatform.tsx [hunk 1]
```diff
@@ -305,7 +303,18 @@ export default class ElectronPlatform extends VectorBasePlatform {
...
+            logger.log("Failed to send setSpellCheckEnabled IPC to Electron");
+            logger.error(error);
+        });
+    }
+
+    public async getSpellCheckEnabled(): Promise<boolean> {
+        return this.ipc.call('getSpellCheckEnabled');
+    }
+
+    setSpellCheckLanguages(preferredLangs: string[]): void {
```
Comment (t3chguy) [inline src/vector/platform/ElectronPlatform.tsx:317]: this lost a `public`

```suggestion
    public setSpellCheckLanguages(preferredLangs: string[]): void {
```

**Review by t3chguy — APPROVED** [review]
LGTM otherwise

**Pull request merged**